### PR TITLE
support for FFTW

### PR DESCRIPTION
**Pull request by MarDiehl**
@conda-forge-admin, please rerender

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged. (nothing has changed)

### CODE REVIEW

**Review by dalcinl — APPROVED** [review]

**Comment by dalcinl** [comment]
@MarDiehl LGTM. Please review a few build logs to double check. Ping me back once ready and I'll merge.

**Comment by dalcinl** [comment]
@MarDiehl Perhaps you need to specify a variant string to use MPI-enabled FFT builds?

**Comment by dalcinl** [comment]
Or perhaps via `conda_build_config.yml`, as done with `hdf5` for the MPI variant.

**Comment by MarDiehl** [comment]
> Or perhaps via `conda_build_config.yml`, as done with `hdf5` for the MPI variant.

I think this is the correct way. I'll give it a try

File: recipe/conda_build_config.yaml
Comment (dalcinl) [inline recipe/conda_build_config.yaml:12]: Bad copy and paste, it should be `fftw`. Also, please add a blank line before.

Comment (MarDiehl) [reply recipe/conda_build_config.yaml:12]: > Bad copy and paste, it should be `fftw`. Also, please add a blank line before.

thx.
Do I also need an equals sign (- = mpi*)? I'm just trying

Comment (dalcinl) [reply recipe/conda_build_config.yaml]: I think `fftw-mpi` is not correct, there is no such package in conda-forge. About the config spec, perhaps `*=mpi*` will do? Sorry, but I'm not really an expert, I'm here since the beginning, and rules have changed so much over time to make sense of things.

**Comment by dalcinl** [comment]
@isuruf Could you please tell us what's the proper way to make this recipe build and depend on the MPI-enabled FFTW package?

File: recipe/meta.yaml
Comment (isuruf) [inline recipe/meta.yaml:59]:
```suggestion
    - fftw {{ fftw }} mpi*
```

File: recipe/meta.yaml
Comment (isuruf) [inline recipe/meta.yaml:59]:
```suggestion
    - fftw {{ fftw }} mpi_{{ mpi }}_*  # [mpi != 'nompi']
```

Comment (MarDiehl) [reply recipe/meta.yaml:59]: thanks!
should this change be done for `host` and `run` or only for `host`?

Comment (isuruf) [reply recipe/meta.yaml:59]: Only needed for host, but doing it for both is fine too.

**Comment by MarDiehl** [comment]
Looks ok now, the only failed test is the result of a timeout.
Anything do do from my side?

File: recipe/meta.yaml
Comment (isuruf) [reply recipe/meta.yaml:59]:
```suggestion
    - fftw {{ fftw }}.* mpi_{{ mpi }}_*  # [mpi != 'nompi']
```

Comment (MarDiehl) [reply recipe/meta.yaml]: does not work

**Comment by dalcinl** [comment]
@MarDiehl Let's start fresh, revert all of your changes to the `meta.yaml` and `conda_build_config.yaml`, and specify the fftw package the following way

```yaml
requirements:
  host:
    - fftw * *mpi_{{ mpi }}*
  run:
    - fftw * *mpi_{{ mpi }}*
```

**Comment by MarDiehl** [comment]
> @MarDiehl Let's start fresh, revert all of your changes to the `meta.yaml` and `conda_build_config.yaml`, and specify the fftw package the following way
> 
> ```yaml
> requirements:
>   host:
>     - fftw * *mpi_{{ mpi }}*
>   run:
>     - fftw * *mpi_{{ mpi }}*
> ```

unfortunately, it does not work.
Do you have any objection against the last working version? I would then put it into a clean branch or force push (not sure if that works)

File: recipe/meta.yaml
Comment (dalcinl) [inline recipe/meta.yaml:59]: I do not understand why we need `# [mpi != 'nompi']`. The `nompi` case simply cannot happen for this recipe, just look at `conda_build_config.yaml`.

@MarDiehl Did you try without the `# [mpi != 'nompi']` comment and think did not work?

**Comment by dalcinl** [comment]
> unfortunately, it does not work.
> Do you have any objection against the last working version?

Could you  clarify? You latest push looks exactly like my last suggestion, plus the `# [ mpi != 'nompi']` comment I do not understand why we need it.

**Comment by MarDiehl** [comment]
> > unfortunately, it does not work.
> > Do you have any objection against the last working version?
> 
> Could you clarify? You latest push looks exactly like my last suggestion, plus the `# [ mpi != 'nompi']` comment I do not understand why we need it.

I copied your suggestion and it failed. But currently we have `fftw * mpi_{{ mpi }}_*  # [mpi != 'nompi']` while your suggestion was `fftw * *mpi_{{ mpi }}*` which differs not only with respect to the comment but also with respect to double `*` and trailing `_`.
I'm new to Conda and only have a rough understanding what the code does. Should I try the current version and just strip the comment?

**Comment by dalcinl** [comment]
> I copied your suggestion and it failed. But currently we have `fftw * mpi_{{ mpi }}_* # [mpi != 'nompi']` while your suggestion was `fftw * *mpi_{{ mpi }}*` which differs not only with respect to the comment but also with respect to double `*` and trailing `_`.

Oh, sorry, yes! My bad, the leading `*` and trailing `_` should not be there.

> I'm new to Conda and only have a rough understanding what the code does.

You are not alone! I also barely know the low level details. It is really frustrating to be a recipe maintainer. I'm truly incompetent. I asked more than once for help (to conda-forge outsiders), but evidently those I asked help for are way smarter than I am.

> Should I try the current version and just strip the comment?

Can we please try without the comment? It should not be required, I'm confident it should work. I do not want confusing leftovers in the recipe, because maintenance is already hard for me, not because of time, but because of understanding.

**Comment by MarDiehl** [comment]
works without the dubious `# [mpi != 'nompi']`.

**Comment by dalcinl** [comment]
@MarDiehl Many thanks for your help and patience!

**Comment by MarDiehl** [comment]
> @MarDiehl Many thanks for your help and patience!

Thank you, for the feedstock and for PETSc in general!

**Pull request merged**
